packages/tanstack-react-query/test/mutationOptions.test.tsx

```diff
@@ -40,122 +40,181 @@ const testContext = () => {
+      const { useTRPC } = ctx;
 
-    const calls: string[] = [];
+      const calls: string[] = [];
 
-    function MyComponent() {
-      const trpc = useTRPC();
+      function MyComponent() {
+        const trpc = useTRPC();
 
-      const options = trpc.post.create.mutationOptions({
-        onMutate(variables) {
-          expectTypeOf<{ text: string }>(variables);
-          calls.push('onMutate');
-        },
-        onSettled(data) {
-          expectTypeOf<'__mutationResult' | undefined>(data);
-          calls.push('onSettled');
-        },
-        onError(_error) {
-          calls.push('onError');
-        },
-        onSuccess(data) {
-          expectTypeOf<'__mutationResult'>(data);
-          calls.push('onSuccess');
-        },
-      });
-      expect(options.trpc.path).toBe('post.create');
+        const options = trpc.post.create.mutationOptions({
+          onMutate(variables) {
+            expectTypeOf<{ text: string }>(variables);
+            calls.push('onMutate');
+          },
+          onSettled(data) {
+            expectTypeOf<'__mutationResult' | undefined>(data);
+            calls.push('onSettled');
+          },
+          onError(_error) {
+            calls.push('onError');
+          },
+          onSuccess(data) {
+            expectTypeOf<'__mutationResult'>(data);
+            calls.push('onSuccess');
+          },
+        });
+        expect(options.trpc.path).toBe('post.create');
 
-      const mutation = useMutation(options);
+        const mutation = useMutation(options);
 
-      React.useEffect(() => {
-        mutation.mutate({
-          text: 'hello',
-        });
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-      }, []);
-      if (!mutation.data) {
-        return <>...</>;
-      }
+        React.useEffect(() => {
+          mutation.mutate({
+            text: 'hello',
+          });
+          // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, []);
+        if (!mutation.data) {
+          return <>...</>;
+        }
 
-      type TData = (typeof mutation)['data'];
-      expectTypeOf<TData>().toMatchTypeOf<'__mutationResult'>();
+        type TData = (typeof mutation)['data'];
+        expectTypeOf<TData>().toMatchTypeOf<'__mutationResult'>();
 
-      return <pre>{JSON.stringify(mutation.data ?? 'n/a', null, 4)}</pre>;
-    }
+        return <pre>{JSON.stringify(mutation.data ?? 'n/a', null, 4)}</pre>;
+      }
+
+      const utils = ctx.renderApp(<MyComponent />);
+      await vi.waitFor(() => {
+        expect(utils.container).toHaveTextContent(`__mutationResult`);
+      });
 
-    const utils = ctx.renderApp(<MyComponent />);
-    await vi.waitFor(() => {
-      expect(utils.container).toHaveTextContent(`__mutationResult`);
+      expect(calls).toEqual(['onMutate', 'onSuccess', 'onSettled']);
     });
 
-    expect(calls).toEqual(['onMutate', 'onSuccess', 'onSettled']);
-  });
+    test('useMutation with no opts', async () => {
```

Comment:
This is the failing case. opts might not be set but our internal types taking from the proxy were assuming they always will, so `opts.keyPrefix` was erroring